apps/server/src/shared/infra/identity-management/keycloak/controller/keycloak-management.controller.ts

```diff
@@ -0,0 +1,25 @@
+import { Controller, ServiceUnavailableException, Post } from '@nestjs/common';
+import { KeycloakManagementUc } from '../uc/Keycloak-management.uc';
+
+@Controller('management/idm')
+export class KeycloakManagementController {
+	constructor(private keycloakManagementUc: KeycloakManagementUc) {}
+
+	/**
+	 * This connects to IDM and seeds the test users.
+	 *
+	 * @returns The number of seeded users
+	 * @throws ServiceUnavailableException if IDM is not ready.
+	 */
+	@Post('seed')
+	async importSeedData(): Promise<number> {
```

Comment:
Good catch! The management controller is used solely for use within the Kubernetes cluster. Hence, seeding is sufficient here. There is no use case where check or clean is used in the auto deployment. Also, the management module (including the controller) is not part of the server.module.ts (public), but only the management.module (no external access).

Reply:
By default all endpoints are blocked by the firewall. But just in case that policy ever changes (e.g. feathers endpoints are blocked by default, but nest endpoints are open) this should better be protected.

I heard that the whole db seeding module is not deployed in prod and thus the endpoint doesn't even exist. But I can't find a feature flag for code for this.

Reply:
@pkleybolte Discussed the matter with SRE team. By default (Kubernetes, Ingress) everything is whitelisted explicitly. Since this is not part of 'server.module.ts' nor a route exits in 'x_ingress.yml' no chance that this is reachable from the outside the cluster.